content/blog/2019-09-23-grpc-on-dotnetcore.md

```diff
@@ -0,0 +1,61 @@
+---
+author: Sourabh Shirhatti
+author-link: https://twitter.com/sshirhatti
+date: "2019-09-23T00:00:00Z"
+published: true
+title: .NET Core ❤ gRPC
+url: blog/grpc-on-dotnetcore
+---
+
+_This is a guest post by [Sourabh Shirhatti](https://twitter.com/sshirhatti), a Program Manger on the .NET team at Microsoft._
+
+The .NET team at Microsoft has been working in close collaboration with the gRPC team since November 2018 on a new fully managed implementation of gRPC for .NET Core.
+
+We're pleased to announce that **grpc-dotnet** is now available with .NET Core 3.0 today!
+
+## How to get it?
+
+**grpc-dotnet** packages have just been released to [NuGet.org](https://www.nuget.org/profiles/grpc-packages) and are already available for use in your projects. These packages also require the latest .NET Core 3.0 shared framework. You can download the .NET Core 3.0 SDK for your dev machine and build servers from the [.NET Core 3.0 download page](https://aka.ms/netcore3download) to acquire the shared framework. 
+
+## Getting Started
+
+gRPC templates are included as part of the .NET SDK. To get started, navigate to a console window after installing the SDK and run the following commands.
+
+```sh
+dotnet new grpc -o GrpcGreeter
+cd GrpcGreeter
+dotnet run
+```
+
+To create a gRPC client and test with the newly created gRPC Greeter service, you can [follow the rest of this tutorial here](https://docs.microsoft.com/aspnet/core/tutorials/grpc/grpc-start).
+
+## Doesn't gRPC already work with .NET Core?
+
+There are currently two official implementations of gRPC for .NET:
+
+- [**Grpc.Core**](https://github.com/grpc/grpc/tree/master/src/csharp): The original gRPC C# implementation based on the native gRPC Core library.
+- [**grpc-dotnet**](https://github.com/grpc/grpc-dotnet): The new implementation written entirely in C# with no native dependencies and based on the newly released .NET Core 3.0.
+
+The implementations coexist side-by-side and each has its own advantages in terms of available features, integrations, supported platforms, maturity level and performance. Both implementations share the same API for invoking and handling RPCs, thus limiting the lock-in and enabling users to choose the implementation that satisfies their needs the best.
+
+##  What's new?
+
+Unlike the existing C-Core based implementation ([Grpc.Core](https://github.com/grpc/grpc/tree/master/src/csharp)), the new libraries ([grpc-dotnet](https://github.com/grpc/grpc-dotnet)) make use of the existing networking primitives in the .NET Core Base Class Libraries (BCL). The diagram below highlights the difference between the existing **Grpc.Core** library and the new **grpc-dotnet** libraries.
+
+<p><img src="/img/grpc-dotnet.svg" alt="gRPC .NET Stack" style="max-width: 800px" /></p>
```

Comment:
I think the image is slightly less informative than it could be, but fair enough.

Optional suggestions:
- make it clear that the HTTP/2 support in grpc-dotnet  comes from ASP.NET Core 
- the "gRPC C-Core Server + Interop layer"  could be "gRPC C-Core native library"
- it's not fully clear from the picture that gRPC C-core actually implements the HTTP/2 stack (not necessary to know, but maybe useful to know).

but also fine as is, maybe it's a question of taste.

Reply:
I tried to make it as similar to https://grpc.io/blog/grpc-stacks/ as possible. I don't want to change it and inundate folks with information. I figured if you're familiar with gRPC, you probably are already acquainted with how the C-Core server works